### PR TITLE
Try to recreate VBO after device rebuild in GS::VertexBuffer

### PR DESCRIPTION
**Pull request by borzun**
In case when can't create VBO in GS::VertexBuffer, retry again after rebuilding device (`gs_present`).

### CODE REVIEW

**Comment by borzun** [comment]
Blocked by https://github.com/stream-labs/obs-studio/pull/190 changes

**Review by summeroff — APPROVED** [review]

**Review by EddyGharbi — APPROVED** [review]

**Comment by EddyGharbi** [comment]
Approved, will merge once OBS is updated to the latest.

**Pull request merged**
